kafl_fuzzer/common/config/settings.py

```diff
@@ -0,0 +1,210 @@
+        raise FileNotFoundError(f"Path {p} doesn't exist")
+    # return string and not PosixPath, since this object is not serializable
+    return str(p)
+
+def cast_expand_path_no_verify(parameter: Any) -> Optional[str]:
+    if parameter is None:
+        return None
+    exp_str = os.path.expandvars(parameter)
+    return exp_str
+
+# register validators
+settings.validators.register(
+    # general
+    Validator("workdir", must_exist=True, cast=cast_expand_path_no_verify),
+    Validator("work_dir", must_exist=True, default=lambda config, _validator: config.workdir),
+    Validator("purge", default=False, cast=bool),
+    Validator("resume", default=False, cast=bool),
+    Validator("processes", cast=int),
+    Validator("verbose", default=False, cast=bool),
+    Validator("quiet", default=False, cast=bool),
+    Validator("log", default=False, cast=bool),
+    Validator("debug", default=False, cast=bool),
+    # fuzz
+    Validator("seed_dir", default=None, cast=cast_expand_path),
+    Validator("dict", default=None, cast=cast_expand_path),
+    Validator("funky", default=False, cast=bool),
+    Validator("afl_dump_mode", default=False, cast=bool),
+    Validator("afl_skip_zero", default=False, cast=bool),
+    Validator("afl_skip_range", default=None),
+    Validator("afl_arith_max", cast=int),
+    Validator("radamsa", default=False, cast=bool),
+    Validator("redqueen", default=False, cast=bool),
+    Validator("redqueen_hashes", default=False, cast=bool),
+    Validator("redqueen_hammer", default=False, cast=bool),
+    Validator("redqueen_simple", default=False, cast=bool),
+    Validator("cpu_offset", cast=int),
+    Validator("abort_time", default=None),
+    Validator("abort_exec", default=None),
+    Validator("timeout_soft", cast=float),
+    Validator("timeout_check", default=False, cast=bool),
+    Validator("kickstart", cast=int),
+    Validator("radamsa_path", default=None, cast=cast_expand_path),
+    # qemu
+    Validator("qemu_image", default=None, cast=cast_expand_path),
+    Validator("qemu_snapshot", default=None, cast=cast_expand_path),
+    Validator("qemu_bios", default=None, cast=cast_expand_path),
+    Validator("qemu_kernel", default=None, cast=cast_expand_path),
+    Validator("qemu_initrd", default=None, cast=cast_expand_path),
+    Validator("qemu_append", default=None),
+    Validator("qemu_memory", cast=int),
+    Validator("qemu_base"),
+    Validator("qemu_serial", default=None),
+    Validator("qemu_extra", default=None),
+    Validator("qemu_path", cast=cast_expand_path),
+    Validator("ip0", default=None, cast=cast_ip_range_to_list),
+    Validator("ip1", default=None, cast=cast_ip_range_to_list),
+    Validator("ip2", default=None, cast=cast_ip_range_to_list),
+    Validator("ip3", default=None, cast=cast_ip_range_to_list),
+    Validator("sharedir", default=None, cast=cast_expand_path),
+    # cannot really map 2 settings to a single value
+    # use a second Validator + default value instead
```

Comment:
we're replacing one hack with another one here...maybe just revert this commit and keep the ['reload'] option?

Reply:
As you wish :-)